### PR TITLE
Donuts setup

### PR DESCRIPTION
**Pull request by eschwelling**
Set up the basic files/alterations for the "donuts" feature (table, controller, model, route).  *User authentication still in the process of setup.

### CODE REVIEW

File: app/controllers/donuts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class DonutsController < ApplicationController
+  def index
+    @donuts=Donuts.all
```
Comment (dpickett) [inline app/controllers/donuts_controller.rb]: spaces before and after assignment here

File: app/models/donut.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Donut < ApplicationRecord
+  CATEGORIES = []
```
Comment (dpickett) [inline app/models/donut.rb]: this should be omitted if the constant will not be used

File: spec/models/donut_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+require_relative "../../app/models/donut.rb"
```
Comment (dpickett) [inline spec/models/donut_spec.rb]: you should not have to require_relative as the `rails_helper` will do this for you :-)

File: app/models/donut.rb [hunk 4]
```diff
@@ -1,11 +1,5 @@
 class Donut < ApplicationRecord
```
Comment (dpickett) [inline app/models/donut.rb]: ok there should be a `belongs_to` association here as well as a `Category` model

Comment (eschwelling) [reply app/models/donut.rb]: addressed in latest commit

File: app/controllers/categories_controller.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class CategoriesController < ApplicationRecord
```
Comment (dpickett) [inline app/controllers/categories_controller.rb]: I think file should be removed. This confused concerns of a controller and model.

File: spec/models/category_spec.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe Category, :type => :model do
```
Comment (dpickett) [inline spec/models/category_spec.rb]: should the name also be unique?

Comment (eschwelling) [reply spec/models/category_spec.rb]: def! updated validations and model test to check for uniqueness.

**Comment by adam-j-curtis** [comment]
Okay, I'm a little nervous saying it . . .  but I think we're ready for another review.  May it pusheth!

**Pull request merged**